### PR TITLE
More pure 5: gregor / login

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers more pure in gregor and login

### CODE REVIEW

File: shared/actions/gregor.js [hunk 1]
```diff
@@ -110,7 +110,8 @@ function registerGregorListeners() {
 }
 
 function* handleTLFUpdate(items: Array<Types.NonNullGregorItem>): SagaGenerator<any, any> {
-  const seenMsgs: Types.MsgMap = yield Saga.select((state: TypedState) => state.gregor.seenMsgs)
+  const state: TypedState = yield Saga.select()
```
Comment (MarcoPolo) [inline shared/actions/gregor.js]: This could be pure

Comment (MarcoPolo) [reply shared/actions/gregor.js]: You'd just change the below to be:
```

function handlePushState(action:..., state) {
...
return Saga.sequentially([
  handleTLFUpdate(nonNullItems, state)
  ...
]) 
```

Comment (MarcoPolo) [reply shared/actions/gregor.js]: (assuming you don't need to see the latest state)

Comment (chrisnojima) [reply shared/actions/gregor.js]: i didn't cause this is actually called by another saga below. and while i could make that work that'd be less of a simple refactor so i avoided it

Comment (chrisnojima) [reply shared/actions/gregor.js]: (which is your suggestion), sorry i got confused about it. still my goal isn't to apply pure to 100% of the things. just the simplest stuff. hopefully with the explorations in devices (the discussion one) we'd eliminate a bunch of this type of code anyways

**Comment by chrisnojima** [comment]
@MarcoPolo is it ok for me to merge this?

**Pull request merged**